core/arch/arm/kernel/spmc_sp_handler.c

```diff
@@ -579,6 +579,72 @@ static void ffa_mem_retrieve(struct thread_smc_args *args,
 	ffa_set_error(args, ret);
 }
 
+static void ffa_mem_relinquish(struct thread_smc_args *args,
+			       struct sp_session *caller_sp,
+			       struct ffa_rxtx  *rxtx)
+{
+	struct sp_mem *smem = NULL;
+	struct ffa_mem_relinquish *mem = rxtx->rx;
+	struct sp_mem_receiver *receiver = NULL;
+	TEE_Result err = FFA_OK;
+	uint32_t exceptions = 0;
+
+	if (!check_rxtx(rxtx)) {
+		ffa_set_error(args, FFA_DENIED);
+		return;
+	}
+
+	cpu_spin_lock(&rxtx->spinlock);
+	smem = sp_mem_get(mem->handle);
+
+	if (!smem) {
+		DMSG("Incorrect handle");
+		err = FFA_DENIED;
+		goto out_spinlock;
+	}
+
+	if (mem->endpoint_count != 1) {
```

Comment:
We should use `READ_ONCE()` here and below with `mem->endpoint_id_array[0]`.

Reply:
Done